### ISSUE TITLE
Doxygen link generation does not find concepts unless complete path is specified

### ISSUE THREAD

**Issue by higaski** [doxygen/doxygen#10395]
For some reason Doxygen seems to make a difference between structs/classes and concepts when it comes to name lookup for link generation.

In the following snippet for example Doxygen is able to create a link to the plain struct but not the concept.

```cpp
namespace c {

/// \page LinkConcept
/// Link to NoConcept works, but link to Concept does not!

struct NoConcept {};

template<typename T>
concept Concept = requires(T a) {
  { std::hash<T>{}(a) } -> std::convertible_to<std::size_t>;
};

}  // namespace c
```

![doxygen_concepts_issue](https://github.com/doxygen/doxygen/assets/13762516/54932c61-0b22-400b-8487-f7421b60f0f8)

The only way I found to get this to work is to explicitly link it using the complete path `\ref c:Concept`

I'm currently on version 1.9.8. For convenience, here's the main.cpp containing the code and the doxyfile:
[doxygen_concepts_issue.zip](https://github.com/doxygen/doxygen/files/13267221/doxygen_concepts_issue.zip)

**Comment by doxygen** [comment]
@higaski Please verify if the referenced commit fixes the problem for you. Do not close this issue, this will be done automatically when the next official release is made available.

**Comment by higaski** [comment]
Looks good, thanks for the fast fix!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
